### PR TITLE
[Export Artifacts] Minor bug fixes and refactoring

### PR DESCRIPTION
**Pull request by niveathika**
## Purpose
$subject
1. Improve regex to identify templated variables
2. Disable export artifacts if there are no siddhi apps in workspace
3. Change shortcut keys to Alt_D and Alt+K. Fixes https://github.com/siddhi-io/distribution/issues/366
4. Disable step link clickable feature in the form flow
5. Keep previous button always to the left
Screenshot from 2019-09-09 13-51-49
6. Use common error alerts format

## Documentation
N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/dialog/export-dialog.js [hunk 1]
```diff
@@ -29,46 +29,58 @@ define(['require', 'jquery', 'log', 'backbone', 'smart_wizard', 'siddhiAppSelect
...
+                    this._btnExportForm =  $('' +
+                        '<form id="submit-form" method="post" enctype="application/x-www-form-urlencoded" target="export-download" >' +
+                        '<button  type="button" class="btn btn-default hidden" id="export-btn" data-dismiss="modal" >Export</button>' +
+                        '</form>').attr('action', exportUrl);
+
                 },
 
                 show: function () {
-                    this.exportContainer.modal('show');
+                    //todo stop form dismiss when clicked away
```
Comment (dnwick) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/dialog/export-dialog.js]: remove todos

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/dialog/export-dialog.js [hunk 2]
```diff
@@ -182,46 +190,46 @@ define(['require', 'jquery', 'log', 'backbone', 'smart_wizard', 'siddhiAppSelect
...
-                    var exportUrl = this.app.config.baseUrl + "/export?type=" + type;
-
-                    this.btnExportForm.append(payload);
-                    this.btnExportForm.attr('action', exportUrl);
-
-                    $(document.body).append(this.btnExportForm);
-                    this.btnExportForm.submit();
+                    $(document.body).append(this._btnExportForm);
+                    this._btnExportForm.submit();
+                    // todo handle error
```
Comment (dnwick) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/dialog/export-dialog.js]: remove todos

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/menu-bar/export-menu.js [hunk 3]
```diff
@@ -28,12 +28,12 @@ define(([], function () {
...
                         mac: {
-                            key: "command+shift+e",
-                            label: "\u2318\u21E7E"
+                            key: "option+d",
+                            label: "\u2325D"
                         },
                         other: {
-                            key: "ctrl+shift+e",
-                            label: "Ctrl+Shift+E"
+                            key: "alt+d",
```
Comment (dnwick) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/menu-bar/export-menu.js]: Please double check whether we have use this combination else where

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/workspace/workspace.js [hunk 4]
```diff
@@ -483,6 +484,34 @@ define(['ace/ace', 'jquery', 'lodash', 'log','dialogs','./service-client','welco
...
+                        if (data.length === 0) {
+                            exportMenuItem.disable();
+                            exportKubeMenuItem.disable();
+                        } else {
+                            exportMenuItem.enable();
+                            exportKubeMenuItem.enable();
+                        }
+                    },
+                    error: function (msg) {
+                        console.info("Error getting siddhi apps: " + msg);
```
Comment (dnwick) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/workspace/workspace.js]: remove console log

Comment (dnwick) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/workspace/workspace.js]: instead add it to a error popup as discussed

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/index.html [hunk 5]
```diff
@@ -2988,6 +2988,7 @@ <h4 class='modal-title export-dialog-title' id='initialHeading'>Exporting Siddhi
                 workspace:  {
                     endpoint: window.location.protocol + "//" + window.location.host + '/editor/workspace'
                 },
+                // todo: deploymentConfigs
```
Comment (dnwick) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/index.html]: remove

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/export-deployment-artifacts/template-file-dialog.js [hunk 6]
```diff
@@ -19,21 +19,22 @@
...
         };
 
         TemplateFileDialog.prototype.constructor = TemplateFileDialog;
 
         TemplateFileDialog.prototype.render = function () {
             var self = this;
             var i;
-            for (i = 0; i < self.appArr.length; i++) {
-                var entry = self.appArr[i];
+            for (i = 0; i < self.appObjectArrayList.length; i++) {
```
Comment (dnwick) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/export-deployment-artifacts/template-file-dialog.js]: better to use loadash js for loops , check in fill template file dialog

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/menu-bar/export-menu.js (hunk 3 above)
Comment (niveathika) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/menu-bar/export-menu.js]: Yes, Alt+D or Alt+K  is not used anywhere

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/index.html (hunk 5 above)
Comment (niveathika) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/index.html]: @pcnfernando FYA

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/export-deployment-artifacts/template-file-dialog.js (hunk 6 above)
Comment (niveathika) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/export-deployment-artifacts/template-file-dialog.js]: @pcnfernando FYA

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/dialog/export-dialog.js (hunk 2 above)
Comment (niveathika) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/dialog/export-dialog.js]: Open an issue for todo https://github.com/siddhi-io/distribution/issues/377

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/workspace/workspace.js (hunk 4 above)
Comment (niveathika) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/workspace/workspace.js]: Fixed in 21cddd8

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/dialog/export-dialog.js (hunk 1 above)
Comment (niveathika) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/js/dialog/export-dialog.js]: Open an issue for todo https://github.com/siddhi-io/distribution/issues/377

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/index.html (hunk 5 above)
Comment (niveathika) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/index.html]: Fixed in f21dc2a

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/export-deployment-artifacts/template-file-dialog.js (hunk 6 above)
Comment (niveathika) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/export-deployment-artifacts/template-file-dialog.js]: Fixed in 5a69e9a

**Pull request merged**